apiserver/facades/controller/instancepoller/instancepoller.go

```diff
@@ -16,10 +16,19 @@ import (
 	corelogger "github.com/juju/juju/core/logger"
 	"github.com/juju/juju/core/network"
 	"github.com/juju/juju/core/status"
+	"github.com/juju/juju/core/watcher"
 	"github.com/juju/juju/rpc/params"
 	"github.com/juju/juju/state"
 )
 
+// MachineService defines the methods that the facade assumes from the Machine
+// service.
+type MachineService interface {
```

Comment:
You don't need any changes to this service other than removal of the watcher logic, since you've injected the service directly into its consuming worker.